parser_library/src/context/using.cpp

```diff
@@ -489,27 +525,43 @@ auto using_collection::using_context::evaluate(id_index label,
 
             result_candidate& target = dist >= 0 ? positive : negative;
             auto abs_dist = std::abs(dist);
-            if (abs_dist < target.min_dist_abs || abs_dist == target.min_dist_abs && reg > target.min_dist_reg)
+            if (auto& t = target.invalid; t.is_better_candidate(abs_dist, reg))
+            {
+                t.min_dist_abs = abs_dist;
+                t.min_dist_reg = reg;
+                t.result_reg = &reg - s.regs.data();
+                t.result_entry = &s;
+                t.min_dist = dist;
+            }
+            if (auto& t = target.valid; t.is_better_candidate(abs_dist, reg) && dist >= min_disp && dist <= max_disp
+                && (ignore_length || fits_limit))
             {
-                target.min_dist_abs = abs_dist;
-                target.min_dist_reg = reg;
-                target.result_reg = &reg - s.regs.data();
-                target.result_entry = &s;
-                target.min_dist = dist;
+                t.min_dist_abs = abs_dist;
+                t.min_dist_reg = reg;
+                t.result_reg = &reg - s.regs.data();
+                t.result_entry = &s;
+                t.min_dist = dist;
```

Comment:
Consider removing the duplicated lines by e.g. concatenating the `if` statements or having a separate function

Reply:
I thought about it, but all the locals are utilized in the expressions and I think it would actually become less clear.